pkg/test/utils/kubernetes.go

```diff
@@ -701,14 +711,13 @@ func WithStatus(t *testing.T, obj runtime.Object, status map[string]interface{})
 }
 
 // WithKeyValue sets key in the provided object to value.
-func WithKeyValue(obj runtime.Object, key string, value map[string]interface{}) (runtime.Object, error) {
-	obj = obj.DeepCopyObject()
-
+func WithKeyValue(obj *unstructured.Unstructured, key string, value map[string]interface{}) (*unstructured.Unstructured, error) {
+	obj = obj.DeepCopy()
+	// we need to convert to and from unstructured here so that the types in case_test match when comparing
```

Comment:
Why do we need the additional convert `ToUnstructured` here? I agree that we need the `FromUnstructured` in Line 723, but it looks like here the object will always be an `Unstructured`, or not? 

I think we could just replace line 717-720 with 
```
content := obj.UnstructuredContent()
```